src/transformers/generation/utils.py

```diff
@@ -2470,8 +2470,11 @@ def sample(
                         else (outputs.hidden_states,)
                     )
 
+            broken_batch_indices = torch.where(torch.max(next_token_scores, dim=-1).values == -torch.inf)[0]
+            _next_token_scores = next_token_scores.index_fill(dim=0, index=broken_batch_indices, value=0.0)
```

Comment:
replace the values at those problematic batch indices with all `0.0`, and assign to a new variable `_next_token_scores` which will be used to compute `probs` below.

We don't change `next_token_scores` because it will be used later to compute the beam scores, and we should not affect any computation other than `probs`.